### PR TITLE
Merge searchSpace service in the HPO

### PR DESCRIPTION
**Pull request by khansaad**
Signed-off-by: Saad Khan <saakhan@redhat.com>

### CODE REVIEW

**Review by dinogun — CHANGES_REQUESTED** [review]
1. Please move tests and scripts outside src
2. no need to have hyperparameter_tuning dir, the python files can be in src

File: scripts/cluster-helpers.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	echo
+	echo "### Installing dependencies.........."
+	echo
+	python3 -m pip install -r requirements.txt
+
+	echo
+	echo "### Starting the service..."
+	echo
+
+	python3 src/service.py
```
Comment (dinogun) [inline scripts/cluster-helpers.sh:89]: In native mode, dont we have to set N_TRIALS and N_JOBS?

File: deploy_hpo.sh [hunk 2]
```diff
@@ -65,6 +66,11 @@ do
 	esac
 done
 
+resolve_container_runtime
+
+echo "Deploying with runtime: ${CONTAINER_RUNTIME}"
+
+
 # Call the proper setup function based on the cluster_type
 if [ ${setup} == 1 ]; then
```
Comment (chandrams) [inline deploy_hpo.sh:75]: We need to set defaults for setup & cluster_type

Comment (khansaad) [reply deploy_hpo.sh:79]: Added default values as 1 and native for seup and clusterType respectively

File: scripts/cluster-helpers.sh (hunk 1 above)
Comment (khansaad) [reply scripts/cluster-helpers.sh:89]: Added default values for env variables

File: src/service.py [hunk 3]
```diff
@@ -28,12 +28,23 @@
...
-host_name = "localhost"
 server_port = 8085
 
+# Default values
+if os.getenv("N_TRIALS") == None :
+    os.environ['N_TRIALS'] = '10'
+
+if os.getenv("N_JOBS") == None :
+    os.environ['N_JOBS'] = '1'
+
```
Comment (dinogun) [inline src/service.py:47]: Can we log these values so that we are aware what are the values for N_JOBS and N_TRIALS

**Comment by chandrams** [comment]
@khansaad - I 'm getting the below error with deploy script in native mode. Can you please check.

Traceback (most recent call last):
  File "src/service.py", line 29, in <module>
    from bayes_optuna import optuna_hpo
  File "/home/csubrama/github/chandra/saad/hpo/src/bayes_optuna/optuna_hpo.py", line 24, in <module>
    n_trials = int(os.getenv("N_TRIALS"))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'

File: src/service.py (hunk 3 above)
Comment (khansaad) [reply src/service.py]: Done

**Review by dinogun — APPROVED** [review]
LGTM

**Comment by dinogun** [comment]
Tested this in native mode, it is working fine now.

**Pull request merged**